crescent/plugin.py

```diff
@@ -9,27 +10,36 @@
 if TYPE_CHECKING:
     from typing import Any, Sequence, TypeVar
 
-    from crescent.typedefs import HookCallbackT
-
-    from .bot import Bot
+    from crescent.bot import Bot
+    from crescent.typedefs import HookCallbackT, PluginCallbackT
 
     T = TypeVar("T", bound="MetaStruct[Any, Any]")
 
 
 __all__: Sequence[str] = ("PluginManager", "Plugin")
 
+_log = getLogger(__name__)
```

Comment:
Since this is a constant it should probably be all-capitals (`_LOG`)

Reply:
> Also I don't think we should keep plugin load/unload hooks in two seperate places. When a MetaStruct is added to a plugin, its hooks should just be added to the Plugin's load/remove hooks imo.

I disagree. Plugin's load and unload hooks have the callback type `Callable[[], None]`. MetaStructs use the callback `Callable[MetaStruct[Any, Any]]` None. The reason MetaStructs can't use an empty callback is so those hook callbacks can be defined wherever (like a function). This isn't an issue for a plugin since the plugin var is already usable in the entire file. Also they are intended for different purposes. The hooks on a plugin are intended for a User's setup and cleanup functions for the _entire_ plugin. MetaStruct's hooks have no bearing on the rest of the plugin and are intended just for itself.

Reply:
Gotcha